Index: libavcodec/mjpegenc.c
===================================================================
--- libavcodec/mjpegenc.c	(revision 10481)
+++ libavcodec/mjpegenc.c	(working copy)
@@ -106,6 +106,7 @@
 
 static void jpeg_table_header(MpegEncContext *s)
 {
+        
     PutBitContext *p = &s->pb;
     int i, j, size;
     uint8_t *ptr;
@@ -198,8 +199,11 @@
 {
     const int lossless= s->avctx->codec_id != CODEC_ID_MJPEG;
 
+
+    goto after_header;
+
     put_marker(&s->pb, SOI);
-
+    
     jpeg_put_comments(s);
 
     jpeg_table_header(s);
@@ -274,6 +278,10 @@
     }
 
     put_bits(&s->pb, 8, 0); /* Ah/Al (not used) */
+
+    after_header:
+    // empty
+    put_marker(&s->pb, SOI);
 }
 
 static void escape_FF(MpegEncContext *s, int start)
Index: libavformat/Makefile
===================================================================
--- libavformat/Makefile	(revision 10481)
+++ libavformat/Makefile	(working copy)
@@ -19,6 +19,7 @@
 OBJS-$(CONFIG_AIFF_MUXER)                += aiff.o riff.o
 OBJS-$(CONFIG_AMR_DEMUXER)               += amr.o
 OBJS-$(CONFIG_AMR_MUXER)                 += amr.o
+OBJS-$(CONFIG_AMV_MUXER)                 += amvenc.o
 OBJS-$(CONFIG_APC_DEMUXER)               += apc.o
 OBJS-$(CONFIG_ASF_DEMUXER)               += asf.o riff.o
 OBJS-$(CONFIG_ASF_MUXER)                 += asf-enc.o riff.o
Index: libavformat/allformats.c
===================================================================
--- libavformat/allformats.c	(revision 10481)
+++ libavformat/allformats.c	(working copy)
@@ -52,6 +52,7 @@
     REGISTER_MUXER    (ADTS, adts);
     REGISTER_MUXDEMUX (AIFF, aiff);
     REGISTER_MUXDEMUX (AMR, amr);
+    REGISTER_MUXER    (AMV, amv);
     REGISTER_DEMUXER  (APC, apc);
     REGISTER_MUXDEMUX (ASF, asf);
     REGISTER_MUXER    (ASF_STREAM, asf_stream);
Index: libavformat/allformats.h
===================================================================
--- libavformat/allformats.h	(revision 10481)
+++ libavformat/allformats.h	(working copy)
@@ -125,6 +125,7 @@
 extern AVOutputFormat adts_muxer;
 extern AVOutputFormat aiff_muxer;
 extern AVOutputFormat amr_muxer;
+extern AVOutputFormat amv_muxer;
 extern AVOutputFormat asf_muxer;
 extern AVOutputFormat asf_stream_muxer;
 extern AVOutputFormat au_muxer;
