Index: libavformat/Makefile
===================================================================
--- libavformat/Makefile	(revision 10455)
+++ libavformat/Makefile	(working copy)
@@ -19,6 +19,7 @@
 OBJS-$(CONFIG_AIFF_MUXER)                += aiff.o riff.o
 OBJS-$(CONFIG_AMR_DEMUXER)               += amr.o
 OBJS-$(CONFIG_AMR_MUXER)                 += amr.o
+OBJS-$(CONFIG_AMV_MUXER)                 += amvenc.o
 OBJS-$(CONFIG_APC_DEMUXER)               += apc.o
 OBJS-$(CONFIG_ASF_DEMUXER)               += asf.o riff.o
 OBJS-$(CONFIG_ASF_MUXER)                 += asf-enc.o riff.o
Index: libavformat/allformats.c
===================================================================
--- libavformat/allformats.c	(revision 10455)
+++ libavformat/allformats.c	(working copy)
@@ -51,6 +51,7 @@
     REGISTER_MUXER    (ADTS, adts);
     REGISTER_MUXDEMUX (AIFF, aiff);
     REGISTER_MUXDEMUX (AMR, amr);
+    REGISTER_MUXER    (AMV, amv);
     REGISTER_DEMUXER  (APC, apc);
     REGISTER_MUXDEMUX (ASF, asf);
     REGISTER_MUXER    (ASF_STREAM, asf_stream);
Index: libavformat/allformats.h
===================================================================
--- libavformat/allformats.h	(revision 10455)
+++ libavformat/allformats.h	(working copy)
@@ -123,6 +123,7 @@
 extern AVOutputFormat adts_muxer;
 extern AVOutputFormat aiff_muxer;
 extern AVOutputFormat amr_muxer;
+extern AVOutputFormat amv_muxer;
 extern AVOutputFormat asf_muxer;
 extern AVOutputFormat asf_stream_muxer;
 extern AVOutputFormat au_muxer;
